### ISSUE TITLE
Review `Collection Header` component for bulk actions and grid layouts

### ISSUE THREAD

**Issue by beatrizmilhomem** [vtex/shoreline#1811]
### Summary

Revisar o header do Collection para que fique mais otimizado e comporte melhor os elementos que já existem e ações em massa.

### Rationale

Durante as discussões do componente de Bulk Actions https://github.com/vtex/shoreline/issues/1766 entendemos que estávamos lidando com um problema maior: o header do Collection. Ficou difícil chegar a um acordo sobre a melhor solução para o Bulk Action porque nenhuma atendia as expectativas para uma usabilidade ótima. Olhando para as principais referências, como Gmail e GitHub, vemos que o header é otimizado, ocupa bem menos espaço e o bulk actions não cria uma área a mais ou causa layout shift.

Por isso, a ideia é revisar o header do Collection de forma que adeque os elementos já existentes e ações em massa.

### Referência
[Bulk Actions](https://www.figma.com/design/FZFazZBtmgUAv2afBQleZ1/Malu's-Sandbox?node-id=2416-145218&t=tIFGzsiWGaif3NQF-4)

### Próximos passos

- [ ] Estudar como o header pode ser mais compacto, levando em consideração bulk actions e grid layouts
- [ ] Testar uso em cenários/produtos distintos

**Comment by malu-viana** [comment]
Concordo! Percebo que seguimos um caminho diferente, de muitos benchmarks que vejo, no uso da `FilterBar`, que acaba ocupando muito espaço e necessita de uma linha da `Collection` só para ela. Também fico com a sensação que a `SearchBar` também poderia ser mais eficiente com o uso do espaço. Sei que ambos os componentes são de uso primário junto na `Collection`, mas no caso da `SearchBar` o usuário precisa de um clique para habilitar o modo de foco, e o mesmo poderia ser alcançado com uma forma mais compacta, como um `Button`. Já o caso dos filtros é mais complexo, de repente possamos nos balizar buscando resolver o cenário de Orders, que é o nosso mais extremo?
